data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java

```diff
@@ -181,11 +193,42 @@ protected CircuitBreakerOptions createCircuitBreakerOptions(
 
   private HttpConsumerRecordSender createSender(
     final String target,
-    final CircuitBreakerOptions circuitBreakerOptions) {
+    final CircuitBreakerOptions circuitBreakerOptions,
+    final EgressConfig egress) {
+
+    final var circuitBreaker = CircuitBreaker.create(target, vertx, circuitBreakerOptions);
+
+    setRetryPolicy(egress, circuitBreaker);
 
     return new HttpConsumerRecordSender(
       client,
-      target
+      target,
+      circuitBreaker
     );
   }
+
+  private void setRetryPolicy(EgressConfig egress, CircuitBreaker circuitBreaker) {
+    if (egress != null) {
+      try {
+        final var delay = Duration.parse(egress.getBackoffDelay()).toMillis();
+
+        final Function<Integer, Long> retryPolicy = switch (egress.getBackoffPolicy()) {
+          case Linear -> retryCount -> linearRetryPolicy(retryCount, delay);
+          case Exponential, UNRECOGNIZED -> retryCount -> exponentialRetryPolicy(retryCount, delay);
+        };
+
+        circuitBreaker.retryPolicy(retryPolicy);
+      } catch (final Exception ex) {
+        logger.error("failed to set retry policy", ex);
```

Comment:
Should this exception be propagated?

Reply:
The only method that can throw the expectation is `parse` and if it fails we have two possible cases:
- It's an empty string (so it's ok to not propagate it)
- It's not a valid ISO 8601 string (so it's a control plane bug)

Reply:
I'm thinking to change the backoff delay's type in our proto in a follow-up PR so we won't have this problem anymore.

Reply:
> It's an empty string (so it's ok to not propagate it)

Why don't you check if the string is empty in the if and throw an IllegalStateException for that?

Reply:
I'll change the type to contain directly milliseconds so we don't have this problem at all, also because we should keep in mind the upstream issue: https://github.com/knative/eventing/issues/4231, can I do this in a separate PR?

Reply:
Ok I'm fine with it